README.md

```diff
@@ -1,6 +1,102 @@
 # Overview
 
-This repository contains software, build flows and examples for the Sonata System (https://github.com/lowRISC/sonata-system)
+This repository contains software, build flows and examples for the Sonata System (https://github.com/lowRISC/sonata-system).
+
+## Installing Nix
+
+The Nix package manager is used to create reproducible builds and consistent development environments.
+We recommend installing Nix with the Determinate Systems' [`nix-installer`](https://github.com/DeterminateSystems/nix-installer):
+
+```sh
+curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install
+```
+
+For more indepth instructions, follow the guide on [the zero to nix site](https://zero-to-nix.com/start/install).
+
+*If you've downloaded nix through another method, make sure the experimental features ["flakes"](https://nixos.wiki/wiki/Flakes) and ["nix-command"](https://nixos.wiki/wiki/Nix_command) are enabled.*
+
+*To use Nix from the terminal you'll need to open up a new terminal for it to be added to your path.*
+
+*To obtain a Linux environment on Windows, you can choose to start a virtual machine or use Windows Subsystem for Linux (WSL). Microsoft provides [a detailed guide on how to install WSL](https://learn.microsoft.com/en-us/windows/wsl/install).*
+
+### Setup Cache
+
+To make use of the lowRISC cache so you don't have to rebuild binaries yourself, you'll need to make sure you're a trusted user.
+To do this, you will need to add your user to the trusted users in `/etc/nix/nix.conf`, e.g. `trusted-users = root username`. *You can also add all users from a certain group instead of a single user by using an `@` symbol before the group name, e.g. `@sudo` or `@wheel`.*
+
+> ℹ️ For Ubuntu users (including WSL users), this means adding this line to the `/etc/nix/nix.conf`:
+> ```
+> trusted-users = root @sudo
+> ```
+> 
+> You'll need to restart the nix-daemon afterwards for the change to be picked up.
+> ```sh
+> sudo systemctl restart nix-daemon
+> ```
+
+## Enter the CHERIoT development environment
+
+Running the following will put you in a shell environment with all the applications needed to build the CHERIoT RTOS.
+
+```sh
+# Enter the shell with
+nix develop github:lowRISC/sonata-software
```

Comment:
```suggestion
nix develop
```

Reply:
I guess this would require them to run this command from within the cloned repository?